src/core/restaurants/getAllRestaurantsSheets.tsx

```diff
@@ -0,0 +1,87 @@
+"use server";
+
+import { get } from "http";
+import { emitKeypressEvents } from "readline";
+import { RestaurantBooleanKeys, RestaurantInfos, restaurantBooleanKeys } from "./interface";
+
+const documentID = process.env.RESTAURANTS_SHEET_ID
+const valuesRange = process.env.SHEET_VALUE_RANGE
+const googleSheetsAPIKey = process.env.GOOGLE_SHEETS_API_KEY
+
+const sheetsColumnsToTechnicalName = {
+  Restaurant: 'name',
+  'Sur place': 'canEatIn',
+  'A emporter': 'canTakeAway',
+  Vege: 'vegetarianFriendly',
+  Viandard : 'meatLover',
+  "URL maps": "mapUrl",
+  "Moins de 10€": "lessThanTenEuros"
+} as const
+
+type GoogleSheetsExpectedColumnNames = keyof typeof sheetsColumnsToTechnicalName
+
+// Correct header type guard
+function isCorrectHeader(header: string[]): header is GoogleSheetsExpectedColumnNames[] {
+  const expectedHeader = Object.keys(sheetsColumnsToTechnicalName)
+  return header.every(col => expectedHeader.includes(col))
+}
+
+// Boolean key type guard
+function isBooleanKey(key: string): key is RestaurantBooleanKeys {
+  return restaurantBooleanKeys.includes(key)
+}
+
+
+const processBoolean = (value: string): boolean => {
+  if (value === 'TRUE'){return true}
+  else if (value === 'FALSE'){return false}
+  else {throw new Error('Value is not in "TRUE" or "FALSE" format')}
+}
+
+const arrayToRestaurantInfos = (data: string[][]): RestaurantInfos[] => {
```

Comment:
```suggestion
const gSheetsDtoToRestaurantInfos = (data: string[][]): RestaurantInfos[] => {
```

array c'est extrêmement générique on ne sait pas c'est quoi

Reply:
done